### PR TITLE
Feat/memory page

### PR DESCRIPTION
**Pull request by wnmay**
- fetch api message
- add phase 1 and 2 of memory page
- fetch user profile in add yours

### CODE REVIEW

**Comment by neennera** [comment]
(neen update) chore : session user.id use google sub instads of google id

File: src/app/Memory/page.tsx [hunk 1]
```diff
@@ -1,16 +1,58 @@
...
+"use client";
 import AddYours from "@/components/memory/AddYours";
-import Banner from "@/components/memory/Banner";
 import { LeftArrow } from "@/components/memory/icon/LeftArrow";
 import { Pencil } from "@/components/memory/icon/Pencil";
 import Link from "next/link";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import CommentSection from "@/components/memory/CommentSection";
+import Phase2Photo from "@/components/memory/Phase2Photo";
```
Comment (phongit-kha) [inline src/app/Memory/page.tsx]: remove `Phase2Photo` because never use

File: src/app/Memory/page.tsx [hunk 2]
```diff
@@ -20,7 +62,7 @@ export default function Memory() {
           <div className="font-vimamsa text-4xl font-normal">Memory</div>
         </div>
         <div className="item-center mx-4 flex justify-center">
-          <img src="https://placehold.co/600x400" />
+          <img src="/memory/larngear_placeholder.webp" />
```
Comment (phongit-kha) [inline src/app/Memory/page.tsx]: using `<Image/>` from next/image

File: src/components/memory/AddYours.tsx [hunk 3]
```diff
@@ -1,127 +1,157 @@
...
+        <div className="relative h-[25.44rem] w-[20.4rem] rounded-lg border-2 border-[#ECF0F6] bg-[#262D37] text-[#ECF0F6]">
+          <button
+            onClick={closeWindow}
+            className="absolute right-2 top-1 text-5xl"
+          >
+            &times;
+          </button>
+
+          <div className="relative left-[2.25rem] top-[4.75rem] mb-2 flex">
+            <img
```
Comment (phongit-kha) [inline src/components/memory/AddYours.tsx]: using `<Image/>` from `next/image`

File: src/components/memory/Phase2Photo.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+          />
+        </div>
+      </div>
+
+      <div className="m-4 text-center font-ibm text-base font-bold">
+        ส่วนกลาง
+      </div>
+      <div className="item-center m-4 flex justify-center">
+        <div className="grid grid-cols-2 gap-4">
+          <img src="https://placehold.co/300x200"></img>
```
Comment (phongit-kha) [inline src/components/memory/Phase2Photo.tsx]: using `<Image/>` from `next/image`

**Comment by phongit-kha** [comment]
fix following this kubb. and ping me after finish kub

### **`./src/app/Memory/page.tsx`**
1. **Line 9:** Remove the unused variable `Phase2Photo`.  
   - Warning: `'Phase2Photo' is defined but never used.`
2. **Line 65:** Replace `` with `<Image />` from `next/image`.  
   - Warning: `Using  could result in slower LCP and higher bandwidth.`
3. **Line 65:** Add an `alt` attribute to `` (or an empty string if decorative).  
   - Warning: `img elements must have an alt prop.`

---

### **`./src/components/memory/AddYours.tsx`**
1. **Line 126:** Replace `` with `<Image />` from `next/image`.  
2. **Line 126:** Add an `alt` attribute to ``.

---

### **`./src/components/memory/Banner.tsx`**
1. **Line 11:** Replace `` with `<Image />` from `next/image`.  
2. **Line 11:** Add an `alt` attribute to ``.

---

### **`./src/components/memory/CommentBox.tsx`**
1. **Line 16:** Replace `` with `<Image />` from `next/image`.  
2. **Line 16:** Add an `alt` attribute to ``.

---

### **`./src/components/memory/Phase2Photo.tsx`**
1. **Line 55:** Replace `` with `<Image />` from `next/image`.  
2. **Line 55:** Add an `alt` attribute to ``.

File: src/components/memory/AddYours.tsx [hunk 5]
```diff
@@ -1,127 +1,155 @@
...
+        description: "Sent! please refresh to see your message",
+        className: "bg-green-500 text-white",
+      });
+      setComment("");
+      close();
+      await timeout(1000);
+      window.location.reload();
+    } catch (error) {
+      console.error("Error:", error);
+      console.log(`${process.env.NEXT_PUBLIC_API_URL}/message`);
```
Comment (phongit-kha) [inline src/components/memory/AddYours.tsx:87]: comment this line

**Review by phongit-kha — APPROVED** [review]

**Pull request merged**